src/mrack/providers/podman.py

```diff
@@ -58,8 +58,58 @@ async def init(self, container_images, default_network, ssh_key, container_optio
         logger.info(f"{self.dsp_name}: Init duration {login_duration}")
 
     async def prepare_provisioning(self, reqs):
-        """Prepare provisioning."""
-        pass
+        """Prepare podman network and pull missing images to prepare provisioning."""
+        image_list = set()
+        network_list = set()
+
+        for req in reqs:
+            # First create list of podman network(s) to be created
+            # podman network(s) should be always created no matter what
+            network = req.get(
+                "network", f"{self.default_network}-{req['domain'].replace('.', '-')}"
+            )
+            # store the used network per requirement to be later accessed
+            req["network"] = network
+            network_list.add(network)
+
+            # Prepare image_list to be pulled later
+            image = req["image"]
+            if not await self.podman.image_exists(image):
```

Comment:
This 'if' should be on line 102 as I wrote earlier.  Where it woudl check it only once per image. 

So that line 100 would be like:
```python
        awaitables = []
        for image in image_list:
            if not await self.podman.image_exists(image):
                 awaitables.append(self.podman.pull(image))
```

Reply:
Updated